app/Config/Publisher.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Config;
+
+use CodeIgniter\Config\Publisher as BasePublisher;
+
+/**
+ * Publisher Configuration
+ *
+ * Defines basic security restrictions for the Publisher class
+ * to prevent abuse by injecting malicious files into a project.
+ */
+class Publisher extends BasePublisher
+{
+	/**
+	 * A list of allowed destinations with a (pseudo-)regex
+	 * of allowed files for each destination.
+	 * Attempts to publish to directories not in this list will
+	 * result in a PublisherException. Files that do no fit the
+	 * pattern will cause copy/merge to fail.
+	 *
+	 * @var array<string,string>
+	 */
+	public $restrictions = [
```

Comment:
How about not default exposing this to userland? Since you have set `Publisher::$restrictions` as private, I think changing should be available only to child Publishers.

Reply:
I thought of making this `protected` with a `__get()` method but ultimately I decided it did not matter. Restrictions are stored during Publisher construction, so an errant module Publisher would not be able to modify the Config value prior to initialization. Any other instances where a module has access to the Config prior to discovery it can already do whatever it wanted anyway.

Reply:
Hmm. Fair enough.

Reply:
I definitely get the suggestion. I went through a bunch of iterations for how to handle security, and basically decided that trying to limit arbitrary module code is useless (without larger framework buy-in) so focused on "what can we do to prevent automated publications from running malicious code?" I believe the current implementation addresses that, with the understanding that there are numerous holes elsewhere in the framework.